src/tools/blast/blastcommandline.jl

```diff
@@ -0,0 +1,170 @@
+# wrapper for BLAST+ command line functions
+
+immutable BlastResult
+    bitscore::Float64
+    expect::Float64
+    queryname::ASCIIString
+    hitname::ASCIIString
+    hit::BioSequence
+    alignment::AlignedSequence
+end
+
+"""
+`readblastXML(blastrun::ASCIIString)`
+Parse XML output of a blast run. Input is an XML string eg:
+```julia
+results = readall(open("blast_results.xml")) # need to use `readstring` instead of `readall` for v0.5
+readblastXML(results)
+```
+
+Returns Vector{BlastResult} with the sequence of the hit, the Alignment with query sequence, bitscore and expect value
+"""
+function readblastXML(blastrun::ASCIIString; seqtype="nucl")
+    xdoc = parse_string(blastrun)
+    xroot = root(xdoc)
+    params = get_elements_by_tagname(xroot, "BlastOutput_param")
+    iterations = get_elements_by_tagname(xroot, "BlastOutput_iterations")
+    results = BlastResult[]
+    for iteration in collect(child_elements(iterations[1]))
+        queryname = content(find_element(iteration, "Iteration_query-def"))
+        hits = get_elements_by_tagname(iteration, "Iteration_hits")
+        for hit in collect(child_elements(hits[1]))
+            hitname = content(find_element(hit, "Hit_def"))
+            hsps = get_elements_by_tagname(hit, "Hit_hsps")
+            for hsp in collect(child_elements(hsps[1]))
+                if seqtype == "nucl"
+                    qseq = BioSequence{DNAAlphabet{4}}(content(find_element(hsp, "Hsp_qseq")))
+                    hseq = BioSequence{DNAAlphabet{4}}(content(find_element(hsp, "Hsp_hseq")))
```

Comment:
I'm going to use `DNASequence` as an alias for the most general DNA sequence type so that it covers 99% of use cases. So I believe `DNASequence` is suitable for the purpose of handling "unseen" sequences returned from BLAST. Do you concur with me, @Ward9250 ?